### PR TITLE
Adds localizedConfiguration

### PR DESCRIPTION
**Pull request by vegaro**
I forgot to add this property that's available in iOS. I noticed when using iOS tests as a base

### CODE REVIEW

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt [hunk 1]
```diff
@@ -35,6 +36,30 @@ data class PaywallData(
...
+     */
+    val localizedConfiguration: LocalizedConfiguration
+        get() {
+            val preferredLocales = LocaleListCompat.getDefault()
+
+            val localizedConfigurations = mutableListOf<LocalizedConfiguration>()
+            for (i in 0 until preferredLocales.size()) {
+                preferredLocales.get(i)?.let { locale ->
+                    configForLocale(locale)?.let { localizedConfiguration ->
+                        localizedConfigurations.add(localizedConfiguration)
```
Comment (tonidero) [inline purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt:50]: Hmm couldn't we just return the moment we find one compatible?

**Review by tonidero — APPROVED** [review]
Just a couple of style comments but LGTM! We will need to confirm that behaves as we want but we can do that later I think.

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt [hunk 2]
```diff
@@ -35,6 +36,30 @@ data class PaywallData(
...
 
+    /**
+     * Returns the [LocalizedConfiguration] to be used based on the current locale.
+     */
+    val localizedConfiguration: LocalizedConfiguration
+        get() {
+            val preferredLocales = LocaleListCompat.getDefault()
+
+            val localizedConfigurations = mutableListOf<LocalizedConfiguration>()
+            for (i in 0 until preferredLocales.size()) {
```
[the remark this answers is not in the document]
Comment (vegaro) [reply purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt:47]: Sadly, there's no `forEach`, it's a `LocaleListCompat`, which is not a collection

[the remark this answers is not in the document]
Comment (tonidero) [reply purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt:47]: Ohh makes sense 😓

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt (hunk 1 above)
Comment (NachoSoto) [reply purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt:50]: Yeah 👍🏻  iOS uses `first(where:)` for that.

File: purchases/src/test/java/com/revenuecat/purchases/paywalls/PaywallDataTest.kt [hunk 3]
```diff
@@ -74,14 +77,51 @@ class PaywallDataTest {
...
-        mockLocale("es_ES", "spa")
         mockLocale("gl_ES", "glg")
 
         val requiredLocale = Locale("gl_ES")
         assertThat(paywall.configForLocale(requiredLocale)).isNull()
     }
 
+    @Test
+    fun `finds locale if it only has language`() {
+        val json = File(javaClass.classLoader!!.getResource(PAYWALLDATA_SAMPLE1).file).readText()
```
Comment (NachoSoto) [inline purchases/src/test/java/com/revenuecat/purchases/paywalls/PaywallDataTest.kt:88]: This is repeated in every test, what do you think about extracting it like `loadJSON(PAYWALLDATA_SAMPLE1)`?

File: purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt [hunk 4]
```diff
@@ -35,6 +36,30 @@ data class PaywallData(
     @SerialName("localized_strings") internal val localization: Map<String, LocalizedConfiguration>,
 ) {
 
+    /**
+     * Returns the [LocalizedConfiguration] to be used based on the current locale.
+     */
+    val localizedConfiguration: LocalizedConfiguration
+        get() {
+            val preferredLocales = LocaleListCompat.getDefault()
```
[the remark this answers is not in the document]
Comment (vegaro) [reply purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt:46]: So far I tested having multiple locales, and there's a list returned of all the locales of the system, in order. In their docs they mention

> The result is guaranteed to include the default Locale returned by Locale.getDefault(), but not necessarily at the top of the list. The default locale not being at the top of the list is an indication that the system has set the default locale to one of the user's other preferred locales, having concluded that the primary preference is not supported but a secondary preference is.

If I understand correctly ` having concluded that the primary preference is not supported ` means the app doesn't support that language, so the language is moved down in the list

[the remark this answers is not in the document]
Comment (tonidero) [reply purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt:46]: 👍 Makes sense

File: purchases/src/test/java/com/revenuecat/purchases/paywalls/PaywallDataTest.kt [hunk 5]
```diff
@@ -74,19 +77,59 @@ class PaywallDataTest {
...
+
+        mockLocale("fr", "fre")
+
+        val enConfig = paywall.configForLocale(Locale("fr"))
+        assertThat(enConfig).isNull()
+    }
+
+    @Test
+    fun `if current locale is missing it loads available locale`() {
+        val json = loadJSON(PAYWALLDATA_MISSING_CURRENT_LOCALE)
```
Comment (NachoSoto) [inline purchases/src/test/java/com/revenuecat/purchases/paywalls/PaywallDataTest.kt:116]: Awesome 👍🏻

**Pull request merged**